### PR TITLE
Partial Update of strain_rates implementation

### PR DESCRIPTION
**Pull request by apcraig**
Rename stress_T and stress_U to stressCD_T and stressCD_U
Rename step[uv]_cgrid to stepuC_[EN], step_vel, stepuCD
create strain_rates_T/U version 2 that operates on arrays, replace most calls to use version 2, still working on a few debugging.  this still needs to be cleaned up.
Add elasticDamp to ice_in
Add a few tests

### CODE REVIEW

File: cicecore/cicedynB/dynamics/ice_dyn_shared.F90 [hunk 1]
```diff
@@ -23,13 +23,15 @@ module ice_dyn_shared
 
       implicit none
       private
-      public :: set_evp_parameters, stepu, step_vel, stepu_Cgrid, stepv_Cgrid, &
+      public :: set_evp_parameters, stepu, stepuCD, stepuC_E, stepuC_N, &
```
Comment (JFLemieux73) [inline cicecore/cicedynB/dynamics/ice_dyn_shared.F90:26]: I would call it stepvC_N.

Comment (JFLemieux73) [reply cicecore/cicedynB/dynamics/ice_dyn_shared.F90:26]: The location (e.g. E) is only given for the C-grid. I would rename the subroutines:
stepu becomes stepuv_B
stepuCD becomes stepuv_CD
stepuC_E becomes stepu_C
stepuC_N becomes stepv_C

File: cicecore/cicedynB/dynamics/ice_dyn_shared.F90 [hunk 2]
```diff
@@ -23,13 +23,15 @@ module ice_dyn_shared
...
                 alloc_dyn_shared, &
                 deformations, deformations_T, &
                 strain_rates, strain_rates_T, strain_rates_U, &
-                shear_strain_rate_U, calc_shearT_DeltaT, &
+                strain_rates_T2, strain_rates_U2, &
+!                shear_strain_rate_U, &
+!                calc_shearT_DeltaT, &
                 visccoeff_replpress, &
                 visccoeff_replpress_avgstr, &
                 visccoeff_replpress_avgzeta, &
```
Comment (eclare108213) [inline cicecore/cicedynB/dynamics/ice_dyn_shared.F90:37]: please change visccoeff to just visc, everywhere it occurs

File: cicecore/cicedynB/dynamics/ice_dyn_shared.F90 (hunk 1 above)
Comment (apcraig) [reply cicecore/cicedynB/dynamics/ice_dyn_shared.F90:26]: Sounds good, I can make those changes (maybe later).  I don't think we should change the names of the B subroutines yet.

**Review by eclare108213 — COMMENTED** [review]
looks like good progress

File: cicecore/cicedynB/dynamics/ice_dyn_shared.F90 [hunk 3]
```diff
@@ -1744,41 +1745,44 @@ subroutine deformations_T (nx_block,   ny_block,   &
...
 
-         call strain_rates_T (nx_block,   ny_block,   &
-                              i,          j,          &
-                              uvelE,      vvelE,      &
-                              uvelN,      vvelN,      &
-                              dxN,        dyE,        &
-                              dxT,        dyT,        &
-                              divT,       tensionT,   &
-                              shearT,     DeltaT      )
+      call strain_rates_T2 (nx_block  ,   ny_block   , &
```
Comment (JFLemieux73) [inline cicecore/cicedynB/dynamics/ice_dyn_shared.F90:1761]: why is it called strain_rates_T2 and not strain_rates_T? Is it just temporary?

**Pull request merged**